Index: ga/GARealGenome.C
===================================================================
--- ga/GARealGenome.C	
+++ ga/GARealGenome.C	
@@ -161,7 +161,7 @@
   if(nMut < 1.0){		// we have to do a flip test on each element
     nMut = 0;
     for(i=length; i>=0; i--){
-      float value = child.gene(i);
+      double value = child.gene(i);
       if(GAFlipCoin(pmut)){
 	if(child.alleleset(i).type() == GAAllele::ENUMERATED ||
 	   child.alleleset(i).type() == GAAllele::DISCRETIZED)
@@ -171,7 +171,7 @@
 	  value = GAMax(child.alleleset(i).lower(), value);
 	  value = GAMin(child.alleleset(i).upper(), value);
 	}
-	child.gene(i, value);
+	child.gene(i, (float)value);
 	nMut++;
       }
     }
@@ -179,7 +179,7 @@
   else{				// only mutate the ones we need to
     for(n=0; n<nMut; n++){
       int idx = GARandomInt(0,length);
-      float value = child.gene(idx);
+      double value = child.gene(idx);
       if(child.alleleset(idx).type() == GAAllele::ENUMERATED ||
 	 child.alleleset(idx).type() == GAAllele::DISCRETIZED)
 	value = child.alleleset(idx).allele();
@@ -188,7 +188,7 @@
 	value = GAMax(child.alleleset(idx).lower(), value);
 	value = GAMin(child.alleleset(idx).upper(), value);
       }
-      child.gene(idx, value);
+      child.gene(idx, (float)value);
     }
   }
   return((int)nMut);
@@ -216,8 +216,8 @@
 
     int len = GAMax(mom.length(), dad.length());
     for(int i=0; i<len; i++) {
-      sis.gene(i, 0.5 * (mom.gene(i) + dad.gene(i)));
-      bro.gene(i, 0.5 * (mom.gene(i) + dad.gene(i)));
+      sis.gene(i, (float)0.5 * (mom.gene(i) + dad.gene(i)));
+      bro.gene(i, (float)0.5 * (mom.gene(i) + dad.gene(i)));
     }
     n = 2;
   }
@@ -228,7 +228,7 @@
 
     int len = GAMax(mom.length(), dad.length());
     for(int i=0; i<len; i++) {
-      sis.gene(i, 0.5 * (mom.gene(i) + dad.gene(i)));
+      sis.gene(i, (float)0.5 * (mom.gene(i) + dad.gene(i)));
     }
     n = 1;
   }
@@ -261,10 +261,10 @@
 	dist = mom.gene(i) - dad.gene(i);
       else 
 	dist = dad.gene(i) - mom.gene(i);
-      float lo = (GAMin(mom.gene(i), dad.gene(i))) - 0.5*dist;
-      float hi = (GAMax(mom.gene(i), dad.gene(i))) + 0.5*dist;
-      sis.gene(i, GARandomFloat(lo, hi));
-      bro.gene(i, GARandomFloat(lo, hi));
+      double lo = (GAMin(mom.gene(i), dad.gene(i))) - 0.5*dist;
+      double hi = (GAMax(mom.gene(i), dad.gene(i))) + 0.5*dist;
+      sis.gene(i, GARandomFloat((float)lo, (float)hi));
+      bro.gene(i, GARandomFloat((float)lo, (float)hi));
     }
     n = 2;
   }
@@ -280,8 +280,8 @@
 	dist = mom.gene(i) - dad.gene(i);
       else 
 	dist = dad.gene(i) - mom.gene(i);
-      float lo = (GAMin(mom.gene(i), dad.gene(i))) - 0.5*dist;
-      float hi = (GAMax(mom.gene(i), dad.gene(i))) + 0.5*dist;
+      float lo = (GAMin(mom.gene(i), dad.gene(i))) - (float)0.5*dist;
+      float hi = (GAMax(mom.gene(i), dad.gene(i))) + (float)0.5*dist;
       sis.gene(i, GARandomFloat(lo, hi));
     }
     n = 1;
Index: ga/garandom.C
===================================================================
--- ga/garandom.C	
+++ ga/garandom.C	
@@ -66,7 +66,7 @@
   if(s == 0 && seed == 0) {
     unsigned long int tmp;
     while(seed == 0) {
-      tmp = time(NULL) _GA_PID;
+      tmp = (unsigned long) time(NULL) _GA_PID;
       for(unsigned int i=0; i<GALIB_BITS_IN_WORD*sizeof(unsigned int); i++)
 	seed += (tmp & (1 << i));
     }
@@ -312,7 +312,7 @@
   iy=iv[j]-idum2;
   iv[j] = idum;
   if (iy < 1) iy += IMM1;
-  if ((temp=AM*iy) > RNMX) return RNMX;
+  if ((temp=(float)AM*iy) > RNMX) return RNMX;
   else return temp;
 }
 
